sympy/tensor/array/sp_utils.py

```diff
@@ -0,0 +1,345 @@
+from sympy.core.sympify import _sympify
+from sympy import Tuple, Basic
+from sympy.tensor.array.ndim_array import NDimArray
+from sympy.tensor.array.sparse_ndim_array import SparseNDimArray
+from sympy.tensor.array.dense_ndim_array import DenseNDimArray
+from sympy.core.compatibility import Iterable
+
+import functools
+
+# A set of functions that could be adde to the base class
+# in order to avoid code repetition
+def check_bound(shape, index):
+    if len(shape) != len(index):
+        return False
+    return all([shape[i] > index[i] for i in range(len(shape))])
+
+# Row-based linked list sparse matrix(LIL)
+class lil(Basic, SparseNDimArray):
```

Comment:
We need a better name.

Reply:
What about LilArray? Or maybe LilSparseArray